tests/tensor_search/integ_tests/test_embed.py

```diff
@@ -0,0 +1,438 @@
+        self.device_patcher.start()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.device_patcher.stop()
+
+    def test_embed_equivalent_to_add_docs(self):
+        for index in [self.unstructured_default_text_index, self.structured_default_text_index]:
+            with self.subTest(index=index.type):
+                add_docs_res = tensor_search.add_documents(
+                    config=self.config, add_docs_params=AddDocsParams(
+                        index_name=index.name,
+                        docs=[
+                            {
+                                "_id": "0",
+                                "text_field_1": "I am the GOAT."
+                            }
+                        ],
+                        device="cpu",
+                        tensor_fields=["text_field_1"] if isinstance(index, UnstructuredMarqoIndex) else None
+                    )
+                )
+
+                # Get the added document embedding
+                get_docs_res = tensor_search.get_document_by_id(
+                    config=self.config, index_name=index.name,
+                    document_id="0", show_vectors=True)
+
+                self.assertEqual(get_docs_res["_id"], "0")
+                self.assertEqual(len(get_docs_res[enums.TensorField.tensor_facets]), 1)
+
+                get_docs_embedding = get_docs_res[enums.TensorField.tensor_facets][0][enums.TensorField.embedding]
+
+                # Embed request the same text
+                embed_res = embed(
+                    marqo_config=self.config, index_name=index.name,
+                    embedding_request=EmbeddingRequest(
+                        content=["I am the GOAT."]
+                    ),
+                    device="cpu"
+                )
+
+                # Assert vectors are equal
+                self.assertEqual(embed_res["content"], ["I am the GOAT."])
+                assert np.allclose(embed_res["embeddings"][0], get_docs_embedding)
+
+    def test_embed_equivalent_to_search_text(self):
+        for index in [self.unstructured_default_text_index, self.structured_default_text_index]:
+            with self.subTest(index=index.type):
+                original_query = self.config.vespa_client.query
+                def pass_through_query(*arg, **kwargs):
+                    return original_query(*arg, **kwargs)
+
+                mock_vespa_client_query = unittest.mock.MagicMock()
+                mock_vespa_client_query.side_effect = pass_through_query
+
+                @unittest.mock.patch("marqo.vespa.vespa_client.VespaClient.query", mock_vespa_client_query)
+                def run():
+                    tensor_search.search(
+                        config=self.config, index_name=index.name, text="I am the GOAT.",
+                        search_method=enums.SearchMethod.TENSOR
+                    )
+                    return True
+                assert run()
```

Comment:
I think using `self.assert(` creates better error messages in case of failure as it calls the test framework. `assert` is a native python instruction. Especially true for equality assertions so applies to all tests here

Reply:
Yep, sorry for missing these! Changing the assert statements.